watertap/unit_models/MD/membrane_distillation_base.py

```diff
@@ -0,0 +1,669 @@
+        )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            doc="Solvent mass flux",
+        )
+        def eq_flux_mass(b, t, x):
+            return b.flux_mass[t, x] == b.permeability_coef[
+                t
+            ] / b.membrane_thickness * (
+                b.hot_ch.properties_interface[t, x].pressure_sat
+                - b.cold_ch.properties_interface[t, x].pressure_sat
+            )
+
+        @self.Expression(
+            self.flowsheet().config.time,
+            doc="Average mass flux expression",
+        )
+        def flux_mass_avg(b, t):
+            return sum(b.flux_mass[t, x] for x in self.difference_elements) / self.nfe
+
+        if (
+            self.config.hot_ch.concentration_polarization_type
+            == ConcentrationPolarizationType.calculated
+        ):
+
+            @self.Constraint(
+                self.flowsheet().config.time,
+                self.difference_elements,
+                solute_set,
+                doc="External Concentration polarization in hot side",
+            )
+            def eq_concentration_polarization_hot_ch(b, t, x, j):
+                jw = b.flux_mass[t, x] / self.dens_solvent
+                exponent = jw / self.hot_ch.K[t, x, j]
+                return b.hot_ch.properties_interface[t, x].conc_mass_phase_comp[
+                    "Liq", j
+                ] == b.hot_ch.properties[t, x].conc_mass_phase_comp["Liq", j] * exp(
+                    exponent
+                )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            doc="hot side evaporation enthalpy flux",
+        )
+        def eq_flux_hot_enth(b, t, x):
+            return (
+                b.flux_enth_hot[t, x]
+                == b.flux_mass[t, x]
+                * b.hot_ch.properties_vapor[t, x].enth_mass_phase["Vap"]
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            doc="cold side evaporation enthalpy flux",
+        )
+        def eq_flux_cold_enth(b, t, x):
+            return (
+                b.flux_enth_cold[t, x]
+                == b.flux_mass[t, x]
+                * b.cold_ch.properties_vapor[t, x].enth_mass_phase["Vap"]
+            )
+
+        @self.Expression(
+            self.flowsheet().config.time,
+            doc="Average hot side enthalpy flux expression",
+        )
+        def flux_enth_hot_avg(b, t):
+            return (
+                sum(b.flux_enth_hot[t, x] for x in self.difference_elements) / self.nfe
+            )
+
+        @self.Expression(
+            self.flowsheet().config.time,
+            doc="Average cold side enthalpy flux expression",
+        )
+        def flux_enth_cold_avg(b, t):
+            return (
+                sum(b.flux_enth_cold[t, x] for x in self.difference_elements) / self.nfe
+            )
```

Comment:
Shouldn't these be consolidated and put in MD_channel_base (or MD_channel_0D)?

Reply:
Mass and heat fluxes connect the channels but are not defined for each individual channel. Even the eq_flux_hot_enth and eq_flux_cold_enth are defined by the mass flux that connects the hot and cold channels.